src/api/AutoBundler.ts

```diff
@@ -81,6 +81,22 @@ export class AutoBundler {
     if (this.fs.existsSync(this.packageJsonFile)) {
       this.processPackageJson(params);
     }
+    else {
+      // If we get here then we've not found a package.json file. it remains possible
+      // that sufficient command line parameters have been set to allow a NODEJSAPP
+      // to be created. If so, process them.
+      if (this.startscriptOverride !== undefined ||
+          this.nodejsappOverride   !== undefined ||
+          this.portOverride        !== undefined ) {
```

Comment:
Why is port required here - can't we still generate the default &HTTP_PORT thing?

Reply:
I figured we shouldn't ignore port if it's set; better to let the validation code complain about the missing startscript/nodejsapp; otherwise the code would just end silently with no error messages.

Reply:
Yes sorry I didn't read it properly!